### PR TITLE
address issues 10

### PR DESCRIPTION
**Pull request by HAOYUatHZ**
https://github.com/KZen-networks/class-groups/blob/master/src/lib.rs#L178

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -46,6 +46,17 @@ pub struct Matrix22 {
     pub a22: BigInt,
 }
 
+impl Matrix22 {
+    pub fn mul(&self, another: Self) -> Self {
```
Comment (amanusk) [inline src/lib.rs]: I could be a good idea to only borrow `another` with `anther: &Self`, in case it needs to be used more that once

**Comment by HAOYUatHZ** [comment]
@amanusk   i have fixed my PR according to ur comments, would u like to take a look again  

couldnt find the "re-request review from" butoon on github

**Comment by amanusk** [comment]
@HAOYUatHZ, thanks, this still does not compile:
``` bash
rror[E0204]: the trait `Copy` may not be implemented for this type
  --> src/lib.rs:42:10
   |
42 | #[derive(Copy, Clone)]
   |          ^^^^
43 | pub struct Matrix22 {
44 |     pub a11: BigInt,
   |     --------------- this field does not implement `Copy`
45 |     pub a12: BigInt,
   |     --------------- this field does not implement `Copy`
46 |     pub a21: BigInt,
   |     --------------- this field does not implement `Copy`
47 |     pub a22: BigInt,
   |     --------------- this field does not implement `Copy`
```

The problem is the `Copy` trait of `BigInt` not `Matrix22`

**Comment by HAOYUatHZ** [comment]
i see
was using i32 for testing my prototype code

sorry for that

**Comment by HAOYUatHZ** [comment]
> @HAOYUatHZ, thanks, this still does not compile:
> 
> ```shell
> rror[E0204]: the trait `Copy` may not be implemented for this type
>   --> src/lib.rs:42:10
>    |
> 42 | #[derive(Copy, Clone)]
>    |          ^^^^
> 43 | pub struct Matrix22 {
> 44 |     pub a11: BigInt,
>    |     --------------- this field does not implement `Copy`
> 45 |     pub a12: BigInt,
>    |     --------------- this field does not implement `Copy`
> 46 |     pub a21: BigInt,
>    |     --------------- this field does not implement `Copy`
> 47 |     pub a22: BigInt,
>    |     --------------- this field does not implement `Copy`
> ```
> 
> The problem is the `Copy` trait of `BigInt` not `Matrix22`

i see
was using i32 for testing my prototype code

sorry for that

**Comment by HAOYUatHZ** [comment]
fixed. can `cargo build` & pass `cargo test`  by using ur @amanusk   https://github.com/KZen-networks/class-groups/pull/18

**Comment by amanusk** [comment]
Looks good to me!

**Pull request merged**